eclair-core/src/main/scala/fr/acinq/eclair/wire/protocol/ChannelTlv.scala

```diff
@@ -107,10 +108,17 @@ object FundingSignedTlv {
   val fundingSignedTlvCodec: Codec[TlvStream[FundingSignedTlv]] = tlvStream(discriminated[FundingSignedTlv].by(varint))
 }
 
-sealed trait FundingLockedTlv extends Tlv
+sealed trait ChannelReadyTlv extends Tlv
 
-object FundingLockedTlv {
-  val fundingLockedTlvCodec: Codec[TlvStream[FundingLockedTlv]] = tlvStream(discriminated[FundingLockedTlv].by(varint))
+object ChannelReadyTlv {
+
+  case class ShortChannelIdTlv(alias: ShortChannelId) extends ChannelReadyTlv
+
+  val channelTypeCodec: Codec[ShortChannelIdTlv] = variableSizeBytesLong(varintoverflow, "alias" | shortchannelid).as[ShortChannelIdTlv]
```

Comment:
The name should be changed:

```suggestion
  val channelAliasCodec: Codec[ShortChannelIdTlv] = variableSizeBytesLong(varintoverflow, "alias" | shortchannelid).as[ShortChannelIdTlv]
```

Also, it would be nice to have tests that verify that we reject the `channel_type`s that use `zero_conf` or `scid_alias` without `anchor_zero_fees`.

Reply:
Renamed in d0a2c97c8, deferring adding test until I convince myself the channel type way is the best.